### DISCUSSION TITLE
Webhook support - need testers

### DISCUSSION THREAD

**Discussion by iklein99** [iklein99/homebridge-smartthings#91] [Announcements]
Hi all.  I have completed work on a new webhook capability which will result in:

* near-realtime state changes into home kit
* Eliminates delays in changes especially motion detectors
* Eliminates polling which reduces network traffic and reduces the risk of hitting API limits with Smartthings.

I will be charging a very small monthly fee for this capability because of the cost of hosting an endpoint in the cloud and the hours I've put into building and testing this capability.  

I am looking for a couple of folks to help me test.  They will have free access to the web hook until I'm ready to launch it into production.  If you are interested in testing, send me an email at iklein99@comcast.net.

**Comment by iklein99** [comment]
Just asking again, does anyone want to help me test this?

**Comment by evlnte** [comment]
Is the solution in production fully locally executed or does it depend on an internet connection. Would this new solution you are proposing here make this plugin internet-dependent?

**Comment by iklein99** [comment]
What I am working on is a version that will, if configured, connect to an external endpoint that will allow Smartthings to send events inbound to your homebridge rather than relying on continuous polling.  Will reduce the network traffic and allow quicker responses to changes (like a motion detector or switch change).  Working on supporting buttons this way as well.

**Reply by sil3n7h** [reply]
If you need any help just let me know. Appreciate all the work on this plugin!!

**Comment by hunkyn** [comment]
@iklein99 i have few devices (Primarily water valves and sensors) in smartthings and I would be happy to test it

**Comment by mikegraben** [comment]
Happy to help test as well @iklein99
